File: modules/client/src/main/java/com/eventhorizon/pulsar/client/packet/processor/BinaryPacketProcessor.java
package com.eventhorizon.pulsar.client.packet.processor;

import com.eventhorizon.pulsar.core.packet.Packet;
import com.eventhorizon.pulsar.core.packet.PacketMetadata;
import org.jetbrains.annotations.NotNull;
import redis.clients.jedis.BinaryJedisPubSub;

public abstract class BinaryPacketProcessor extends BinaryJedisPubSub {

    public abstract void startListening();

    public abstract void publish(@NotNull PacketMetadata metadata, @NotNull Packet packet);
}

File: modules/client/src/main/kotlin/com/eventhorizon/pulsar/client/packet/processor/BinaryPacketProcessor.kt
package com.eventhorizon.pulsar.client.packet.processor

import com.eventhorizon.pulsar.core.packet.Packet
import com.eventhorizon.pulsar.core.packet.PacketMetadata
import redis.clients.jedis.BinaryJedisPubSub

abstract class BinaryPacketProcessor : BinaryJedisPubSub() {
    abstract suspend fun startListening()

    abstract suspend fun publish(metadata: PacketMetadata, packet: Packet)
}